processor/src/main/java/com/hannesdorfmann/fragmentargs/processor/ArgProcessor.java

```diff
@@ -729,12 +751,26 @@ private void scanForAnnotatedFragmentClasses(RoundEnvironment env,
     /**
      * Checks if the given element is in a valid Fragment class
      */
-    private boolean isFragmentClass(Element classElement, TypeElement fragmentType,
-                                    TypeElement supportFragmentType) {
+    private boolean isFragmentClass(Element classElement) {
+        List<TypeElement> fragmentTypeElements = Arrays.asList(TYPE_FRAGMENT, TYPE_SUPPORT_FRAGMENT, TYPE_ANDROIDX_FRAGMENT);
+
+        for (TypeElement fragmentTypeElement : fragmentTypeElements) {
+            if(fragmentTypeElement != null && typeUtils.isSubtype(classElement.asType(),
+                    fragmentTypeElement.asType())) {
+                return true;
+            }
+        }
 
-        return (fragmentType != null && typeUtils.isSubtype(classElement.asType(),
-                fragmentType.asType())) || (supportFragmentType != null && typeUtils.isSubtype(
-                classElement.asType(), supportFragmentType.asType()));
+        return false;
+    }
+
+    private boolean isClassAvailable(String className) {
```

Comment:
Not sure if this works always. Should we check if `TypeElement` exists or is that done someone before calling this and i have overlooked it?

Reply:
Works fine for me but i am still not sure whats the best way to check if a Class exists. I haven't tried using `TypeElement` for it yet. Will investigate and update my PR accordingly.

Reply:
I will take a look at this at during the weekend